### PR TITLE
Unify npm and node versions across environments

### PR DESCRIPTION
**Pull request by chriddyp**
We have 4 environments:
- Docker (Available node versions here: https://hub.docker.com/_/node/). 
  
  Note that we can't install different versions of npm, we have to use the version of npm that was released with the version of node.

- Heroku - NPM and Node are installed with the `engines` variable in `package.json`

- Local dev environments - should also use the `engines` variable in `package.json`

- CircleCI - Unfortunately CircleCI only supports node up to `v6.1.0` (https://circleci.com/docs/build-image-trusty/#nodejs)

I've run into a series of gnarly NPM issues in the last few weeks including:
- https://github.com/npm/npm/issues/14427#issuecomment-264358546
- https://github.com/npm/npm/issues/14042

There are two issues on `master` right now.
- Heroku builds are failing because of one of the issues above ^^
- Our Docker's NPM version is different than our `package.json`. That makes me worried about intermittent docker build failures in the future (like the one that we just suddenly randomly saw from Heroku).

This PR unifies all of the versions of NPM and Node to solve those two issues.

### CODE REVIEW

**Comment by chriddyp** [comment]
- [ ] Tested building docker locally
- [ ] Manually tested heroku deployment
- [ ] Manually tested local deployment
- [ ] CircleCI passes

**Comment by chriddyp** [comment]
@scjody @alexandresobolevski - Can you review real quick?

**Comment by chriddyp** [comment]
(just updated the main comment)

File: CHANGELOG.md [hunk 1]
```diff
@@ -1,3 +1,20 @@
+# 1.0.1 - 2016-12-08
+Updated Node and NPM versions
+
+# 1.0.0 - 2016-12-08
```
Comment (scjody) [inline CHANGELOG.md]: :cow2: It would be great to follow the same Changelog version as we use in streambed: http://keepachangelog.com/en/0.3.0/

**Comment by alexandresobolevski** [comment]
👯

**Pull request merged**